### PR TITLE
chore(rosetta): de-duplicate diagnostics early

### PR DESCRIPTION
**Pull request by RomainMuller**
Making sure diagnostics are always sorted and de-duplciated reduces
the amount of overhead pertaining to keeping diagnostics in memory
and could result in better overall performance (due to lower memory
pressure)

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: packages/jsii-rosetta/lib/translate.ts [hunk 1]
```diff
@@ -44,7 +44,8 @@ export function translateTypeScript(
  */
 export class Translator {
   private readonly compiler = new TypeScriptCompiler();
-  public readonly diagnostics: ts.Diagnostic[] = [];
+  // eslint-disable-next-line @typescript-eslint/explicit-member-accessibility
```
Comment (nija-at) [inline packages/jsii-rosetta/lib/translate.ts:47]: Oh gosh, typescript eslint is not catching up with newer typescript features

File: packages/jsii-rosetta/lib/translate.ts [hunk 2]
```diff
@@ -107,7 +114,7 @@ export interface SnippetTranslatorOptions extends AstRendererOptions {
 
 export interface TranslateResult {
   translation: string;
-  diagnostics: ts.Diagnostic[];
+  diagnostics: readonly ts.Diagnostic[];
```
Comment (nija-at) [inline packages/jsii-rosetta/lib/translate.ts:117]: What does it mean to have `readonly` after the `:`?

Also mark `translation` as `readonly`?

**Review by nija-at — APPROVED** [review]

File: packages/jsii-rosetta/lib/translate.ts (hunk 1 above)
Comment (RomainMuller) [reply packages/jsii-rosetta/lib/translate.ts:47]: The culprit appears to be that Babel does not support this just yet 🤷🏻‍♂️ or not in a stable way (couldn't completely understand what I read on the subject)

File: packages/jsii-rosetta/lib/translate.ts (hunk 2 above)
Comment (RomainMuller) [reply packages/jsii-rosetta/lib/translate.ts:117]: `readonly` here is a modifier applied to the array type... Actually means I promise (under the type checker's supervision) I will not modify the array provided as input.

**Pull request merged**